x-pack/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/nodes/run_agent.ts

```diff
@@ -9,34 +9,48 @@ import { RunnableConfig } from '@langchain/core/runnables';
 import { AgentRunnableSequence } from 'langchain/dist/agents/agent';
 import { AgentState, NodeParamsBase } from '../types';
 import { NodeType } from '../constants';
+import { AIAssistantKnowledgeBaseDataClient } from '../../../../../ai_assistant_data_clients/knowledge_base';
 
 export interface RunAgentParams extends NodeParamsBase {
   state: AgentState;
   config?: RunnableConfig;
   agentRunnable: AgentRunnableSequence;
+  kbDataClient?: AIAssistantKnowledgeBaseDataClient;
 }
 
 export const AGENT_NODE_TAG = 'agent_run';
 
+const KNOWLEDGE_HISTORY_PREFIX = 'Knowledge History:';
+const NO_KNOWLEDGE_HISTORY = '[No existing knowledge history]';
+
 /**
  * Node to run the agent
  *
  * @param logger - The scoped logger
  * @param state - The current state of the graph
  * @param config - Any configuration that may've been supplied
  * @param agentRunnable - The agent to run
+ * @param kbDataClient -  Data client for accessing the Knowledge Base on behalf of the current user
  */
 export async function runAgent({
   logger,
   state,
   agentRunnable,
   config,
+  kbDataClient,
 }: RunAgentParams): Promise<Partial<AgentState>> {
   logger.debug(() => `${NodeType.AGENT}: Node state:\n${JSON.stringify(state, null, 2)}`);
 
+  const knowledgeHistory = await kbDataClient?.getRequiredKnowledgeBaseDocumentEntries();
+
   const agentOutcome = await agentRunnable.withConfig({ tags: [AGENT_NODE_TAG] }).invoke(
     {
       ...state,
+      knowledge_history: `${KNOWLEDGE_HISTORY_PREFIX}\n${
+        knowledgeHistory?.length
+          ? JSON.stringify(knowledgeHistory.map((e) => e.text))
+          : NO_KNOWLEDGE_HISTORY
+      }`,
```

Comment:
nit: I had considered pulling this out to it's own `AddKnowledgeHistory` node (similar to how we fetch chat_history and add to graph state). This would probably be good cleanup